Sprint2/paella

is-a-me opened the pull request:
Sprint 2 - Feature implemented (issue #271)
tools.pdfffontsinfo.py - main code of the feature
It collects the font list from pfd file.

@DanelArias-Dreyton257 & @is-a-me

DanelArias-Dreyton257 committed:
tests for the pdffonts, before implementation

Co-Authored-By: Isa <79759822+is-a-me@users.noreply.github.com>

DanelArias-Dreyton257 committed:
PDFFontsInfo main implementation and rebrabnanding (pdffonts -> pdffontsinfo)

Co-Authored-By: Isa <79759822+is-a-me@users.noreply.github.com>

is-a-me committed:
Changes in pdffont to retrieve the enconding and the toUnicode

Co-Authored-By: Danel Arias <71316377+danelarias-dreyton257@users.noreply.github.com>

is-a-me committed:
Converter adapted and output fixed

Co-Authored-By: Danel Arias <71316377+danelarias-dreyton257@users.noreply.github.com>

is-a-me committed:
Covering funtionality for all samples pdfs

Co-Authored-By: Danel Arias <71316377+danelarias-dreyton257@users.noreply.github.com>

is-a-me committed:
Clear cache and encoding name bug fixed

Co-Authored-By: Danel Arias <71316377+danelarias-dreyton257@users.noreply.github.com>

is-a-me committed:
Tests for sample2 and sample3

Co-Authored-By: Danel Arias <71316377+danelarias-dreyton257@users.noreply.github.com>

DanelArias-Dreyton257 committed:
ReadMe updated with how to pdffontsinfo

ghmo2789 commented on tools/pdffontsinfo.py:
```diff
@@ -0,0 +1,222 @@
+    caching = True
+
+    laparams = LAParams()
+    if args.debug:
+        debug += 1
+    elif args.password:
+        password = args.password.encode('ascii')
+    elif args.output:
+        outfile = args.output
+    elif args.rotation:
+        rotation = int(args.rotation)
+    elif args.pagenos:
+        pagenos.update(int(x)-1 for x in args.pagenos.split(','))
+    elif args.maxpages:
+        maxpages = int(args.maxpages)
+    elif args.disable_caching:
+        caching = False
+    elif args.no_layout:
+        laparams = None
+    elif args.all_texts:
+        laparams.all_texts = True
+    elif args.detect_vertical:
+        laparams.detect_vertical = True
+    elif args.char_margin:
+        laparams.char_margin = float(args.char_margin)
+    elif args.word_margin:
+        laparams.word_margin = float(args.word_margin)
+    elif args.line_margin:
+        laparams.line_margin = float(args.line_margin)
+    elif args.boxes_flow:
+        laparams.boxes_flow = float(args.boxes_flow)
```
Change all the elif to if, otherwise the python will not check for them if the first if-statement is true.

DanelArias-Dreyton257 replied:
New commits have been done following this requested changes. Check them out!

ghmo2789 reviewed (changes requested):
Everything looks great except for a change in the pdffontsinfo.py file. See my comment for detailed. There were also some warnings generate by flake8, go through them also. Appreciate that your code is well commented.

dakaza98 commented on tools/pdffontsinfo.py:
```diff
@@ -0,0 +1,222 @@
+                page.rotate = (page.rotate + rotation) % 360
+                interpreter.process_page(page)
+
+        l_font_names = []
+        l_font_types = []
+        l_font_enc = []
+        l_font_uni = []
```
You can remove these and use the fonts object directly

DanelArias-Dreyton257 replied:
Fixed! Thanks for the suggestion.

dakaza98 commented on tools/pdffontsinfo.py:
```diff
@@ -0,0 +1,222 @@
+            l_font_types.append(font_type)
+            l_font_names.append(font_name)
+            l_font_enc.append(font_enc)
+            l_font_uni.append(font_uni)
```
I also recommend changing the names of `enc` and `uni` to `encoding` and `is_unicode`. It makes the code more readable
```suggestion
            fonts['types'].append(font_type)
            fonts['names'].append(font_name)
            fonts['encoding'].append(font_enc)
            fonts['is_unicode'].append(font_uni)
```

DanelArias-Dreyton257 replied:
Fixed! Thanks for the suggestion.

dakaza98 commented on tools/pdffontsinfo.py:
```diff
@@ -0,0 +1,222 @@
+    ''' Transforms the font information in a nice text table following
+        this pattern:
+
+        Name                       Type     Encoding             Uni
```
I think changing `Uni` to `Unicode` would make the table more understandable for the user when they run the command

DanelArias-Dreyton257 replied:
Fixed! Thanks for the suggestion.

dakaza98 commented on tools/pdffontsinfo.py:
```diff
@@ -0,0 +1,222 @@
+        Helvetica                  Type 1   Custom               no
+        Helvetica-Bold             Type 1   Standard             no
+
+        The table header: 4 titles: 'Name', 'Type', 'Encoding', 'Uni'.
```
Great documentation :tada:

dakaza98 commented on tools/pdffontsinfo.py:
```diff
@@ -0,0 +1,222 @@
+        name_space = 25
+        type_space = 10
+        encoding_space = 20
+        uni_space = 3
```
Changing to unicode makes the code more readable
```suggestion
        unicode_space = 3
```

DanelArias-Dreyton257 replied:
Great suggestion! Changes implemented :D

dakaza98 commented on tools/pdffontsinfo.py:
```diff
@@ -0,0 +1,222 @@
+                - uni: list of booleans that shows if it is unicode or not
+
+    '''
+    def process_line(list_info):
```
I think the code becomes easier to use if you use one argument for each value instead of an array
```suggestion
    def process_line(name, type, encoding, unicode):
```

DanelArias-Dreyton257 replied:
Great suggestion! Changes implemented :D

DanelArias-Dreyton257 committed:
Flake 8 errors

Co-Authored-By: Isa <79759822+is-a-me@users.noreply.github.com>

DanelArias-Dreyton257 committed:
elifs changed to if statements

Avoid the exclusion of elif statements when recieving the arguments

Co-Authored-By: Isa <79759822+is-a-me@users.noreply.github.com>

DanelArias-Dreyton257 committed:
renamed 'enc' and 'uni' for readability

Co-Authored-By: Isa <79759822+is-a-me@users.noreply.github.com>

DanelArias-Dreyton257 committed:
fonts object used directly instead of intermidiate lists

Co-Authored-By: Isa <79759822+is-a-me@users.noreply.github.com>

DanelArias-Dreyton257 committed:
updated processline function following suggestion

Now the func receives 4 params instead of a list and some naming has been changed for more readability

Co-Authored-By: Isa <79759822+is-a-me@users.noreply.github.com>
